python/BioSimSpace/Sandpit/Exscientia/FreeEnergy/_restraint.py

```diff
@@ -311,86 +346,244 @@ def write_dihedral(key_list, equilibrium_values, force_constants):
+        kr = (self._restraint_dict['force_constants']['kr']/(_kcal_per_mol / (_angstrom*_angstrom)))/2
+        kthetaA = (self._restraint_dict['force_constants']['kthetaA']/(_kcal_per_mol / (_radian*_radian)))/2
+        kthetaB = (self._restraint_dict['force_constants']['kthetaB']/(_kcal_per_mol / (_radian*_radian)))/2
+        kphiA = (self._restraint_dict['force_constants']['kphiA']/(_kcal_per_mol / (_radian*_radian)))/2
+        kphiB = (self._restraint_dict['force_constants']['kphiB']/(_kcal_per_mol / (_radian*_radian)))/2
+        kphiC = (self._restraint_dict['force_constants']['kphiC']/(_kcal_per_mol / (_radian*_radian)))/2
+
+        restr_string = f'boresch restraints dictionary = {{"anchor_points":{{"r1":{r1}, "r2":{r2}, "r3":{r3}, "l1":{l1}, '
+        restr_string += f'"l2":{l2}, "l3":{l3}}}, '
+        restr_string += f'"equilibrium_values":{{"r0":{r0:.2f}, "thetaA0":{thetaA0:.2f}, "thetaB0":{thetaB0:.2f},"phiA0":{phiA0:.2f}, '
+        restr_string += f'"phiB0":{phiB0:.2f}, "phiC0":{phiC0:.2f}}}, '
+        restr_string += f'"force_constants":{{"kr":{kr:.2f}, "kthetaA":{kthetaA:.2f}, "kthetaB":{kthetaB:.2f}, "kphiA":{kphiA:.2f}, '
+        restr_string += f'"kphiB":{kphiB:.2f}, "kphiC":{kphiC:.2f}}}}}'
+
+        return restr_string
+
+
+    def toString(self, engine=None):
+        """The method for convert the restraint to a format that could be used
         by MD Engines.
 
         Parameters
         ----------
 
         engine : str
             The molecular dynamics engine used to generate the restraint.
-            Available options currently is "GROMACS" only. If this argument
+            Available options currently is "GROMACS" and "SOMD". If this argument
             is omitted then BioSimSpace will choose an appropriate engine
             for you.
         """
         if engine.strip().lower() == "gromacs":
             if self._restraint_type == "boresch":
                 return self._gromacs_boresch()
+        elif engine.lower() == 'somd':
+            if self._restraint_type == 'boresch':
+                return self._somd_boresch()
             else:
                 raise NotImplementedError(
-                    f"Restraint type {self.restraint_type} not implemented "
-                    f"yet. Only boresch restraint is supported."
-                )
+                    f'Restraint type {self._restraint_type} not implemented '
+                    f'yet. Only Boresch restraints are supported.')
         else:
-            raise NotImplementedError(
-                f"MD Engine {engine} not implemented "
-                f"yet. Only Gromacs is supported."
-            )
+            raise NotImplementedError(f'MD Engine {engine} not implemented '
+                                      f'yet. Only Gromacs and SOMD are supported.')
+
+
+    def getCorrection(self, method="numerical"):
+        """Calculate the free energy of releasing the restraint
```

Comment:
Multi-line docstrings start on a new line.

Reply:
The indentation of the rest of the docstring will need to be adjusted accordingly.